#!/bin/bash

#PBS -l nodes=1:ppn=20,walltime=6:00:00
#PBS -j oe
#PBS -m abe
#PBS -M mdb5946@psu.edu
#PBS -A wff3_a_g_hc_default

# Go to the correct place

cd $PBS_O_WORKDIR

module load python/3.6.8
module load anaconda3
source activate /storage/home/mdb5946/.conda/envs/qiime2-2020.8

qiime feature-classifier classify-sklearn \
  --i-classifier /gpfs/group/dut374/default/marcella/16s2_ny/silva/silva_classifier.qza \
  --i-reads rep-seqs.qza \
  --p-n-jobs 4 \
  --o-classification silva_taxonomy.qza

qiime metadata tabulate \
  --m-input-file silva_taxonomy.qza \
  --o-visualization silva_taxonomy.qzv

#visualize classified sequences
qiime taxa barplot \
  --i-table feature_table.qza \
  --i-taxonomy silva_taxonomy.qza \
  --m-metadata-file metadata_16s_vermivora2.tsv \
  --o-visualization taxa-bar-plots.qzv

#for decontam:

#export feature table to otu table for decontam (puts it in a folder called otu-table)
qiime tools export \
  --input-path feature_table.qza \
  --output-path otu_table_forDecontam

#convert to tsv file for decontam
biom convert -i  ./otu_table_forDecontam/feature-table.biom -o ./otu_table_forDecontam/feature-table.tsv --to-tsv

#generate a tree for decontam
qiime phylogeny align-to-tree-mafft-fasttree \
  --i-sequences rep-seqs.qza \
  --o-alignment aligned-rep-seqs.qza \
  --o-masked-alignment masked-aligned-rep-seqs.qza \
  --o-tree unrooted-tree_forDecontam.qza \
  --o-rooted-tree rooted-tree_forDecontam.qza

#export tree for decontam
qiime tools export \
  --input-path rooted-tree_forDecontam.qza \
  --output-path exported-tree_forDecontam

#don't forget to download tsv of silva_taxonomy.qzv from qiime2 view for decontam!
